drivers/adodb-firebird.inc.php

```diff
@@ -307,23 +435,36 @@ function SelectDB($dbName)
 		return false;
 	}
 
+
 	function _handleerror()
 	{
-		$this->_errorMsg = fbird_errmsg();
+		$this->_errorCode = fbird_errcode();
+		$this->_errorMsg  = fbird_errmsg();
 	}
 
-	function ErrorNo()
+
+	public function errorNo()
 	{
-		if (preg_match('/error code = ([\-0-9]*)/i', $this->_errorMsg,$arr)) return (integer) $arr[1];
-		else return 0;
+		return (integer) $this->_errorCode;
 	}
 
 	function ErrorMsg()
 	{
 			return $this->_errorMsg;
 	}
 
-	function Prepare($sql)
+	/**
+	 * Prepares an SQL statement and returns a handle to use.
+	 * This is not used by bound parameters anymore
+	 *
+	 * @link https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:prepare
+	 * @todo update this function to handle prepared statements correctly
+	 *
+	 * @param string $sql The SQL to prepare.
+	 *
+	 * @return string The original SQL that was provided.
```

Comment:
The function actually returns an array

Reply:
Returns an array, or false if the preparation fails